htlcswitch/circuit_map.go

```diff
@@ -224,6 +229,134 @@ func (cm *circuitMap) initBuckets() error {
 	})
 }
 
+// cleanClosedChannels deletes all circuits and keystones related to closed
+// channels. It first reads all the closed channels and cache the ShortChanIDs
+// into a map for fast lookup. Then it iterates the circuit bucket and keystone
+// bucket and deletes items whose ChanID matches the ShortChanID.
+//
+// NOTE: this operation can also be built into restoreMemState since the latter
+// already opens and iterates the two root buckets, circuitAddKey and
+// circuitKeystoneKey. Depending on the size of the buckets, this marginal gain
+// may be worth investigating. Atm, for clarity, this operation is wrapped into
+// its own function.
+func (cm *circuitMap) cleanClosedChannels() error {
+	log.Infof("Cleaning circuits from disk for closed channels")
+
+	var (
+		targetChanIDs       = make(map[lnwire.ShortChannelID]int)
```

Comment:
Any reason this is mapped to an integer rather than just a `bool` or `struct{}`?

Reply:
Guess I wasn't aware of using empty struct back then🤓